filecache.go

```diff
@@ -198,47 +352,52 @@ func (c *FileCache) GetFileName(filename string) string {
 	extension := c.DefaultExtension
 
 	// Look in the last 5 characters for a . and extension
-	lastDot := strings.LastIndexByte(filename, '.')
-	if lastDot > len(filename)-6 {
-		extension = filename[lastDot:]
+	lastDot := strings.LastIndexByte(downloadRecord.Path, '.')
+	if lastDot > len(downloadRecord.Path)-6 {
+		extension = downloadRecord.Path[lastDot:]
 	}
 
 	file := fmt.Sprintf("%x%s", hashedFilename, extension)
 	dir := fmt.Sprintf("%x", hashedDir[:1])
 	return filepath.Join(c.BaseDir, dir, filepath.FromSlash(path.Clean("/"+file)))
 }
 
-// onEvictDelete is a callback that is triggered when the LRU cache expires an
-// entry.
-func (c *FileCache) onEvictDelete(key interface{}, value interface{}) {
-	filename := key.(string)
-	storagePath := value.(string)
+// bucketToDownloadManager matches the given bucket to a suitable download manager
+// TODO: Implement this in a more robust / generic way
```

Comment:
This won't allow us to use an S3 bucket called "dropbox", but that should be fine for now. I don't want to break the existing HTTP URL API structure in Lazyraster unless really needed.